pkg/ebpf/driver_interface.go

```diff
@@ -0,0 +1,373 @@
+
+// Creates a buffer that Driver will use to verify proper versions are communicating
+// We create a buffer because the system calls we make need a *byte which is not
+// possible with const value
+func makeDDAPIVersionBuffer(signature uint64) []byte {
+	buf := make([]byte, C.sizeof_uint64_t)
+	binary.LittleEndian.PutUint64(buf, signature)
+	return buf
+}
+
+// DriverInterface holds all necessary information for interacting with the windows driver
+type DriverInterface struct {
+	driverFlowHandle  *DriverHandle
+	driverStatsHandle *DriverHandle
+
+	path       string
+	totalFlows int64
+}
+
+// NewDriverInterface returns a DriverInterface struct for interacting with the driver
+func NewDriverInterface() (*DriverInterface, error) {
+	dc := &DriverInterface{
+		path: driverFile,
+	}
+
+	err := dc.SetupFlowHandle()
+	if err != nil {
+		return nil, fmt.Errorf("%s : %s", "error creating driver flow handle", err)
+	}
+
+	err = dc.SetupStatsHandle()
+	if err != nil {
+		return nil, fmt.Errorf("%s : %s", "error creating driver stats handle", err)
+	}
+
+	return dc, nil
+}
+
+func (di *DriverInterface) close() error {
+	err := windows.CloseHandle(di.driverFlowHandle.handle)
+	if err != nil {
+		log.Errorf("error closing flow file handle %v", err)
+	}
+	err = windows.CloseHandle(di.driverStatsHandle.handle)
+	if err != nil {
+		log.Errorf("error closing stat file handle %v", err)
+	}
+	return err
+}
+
+// SetupFlowHandle generates a windows Driver Handle, and creates a DriverHandle struct to pull flows from the driver
+// by setting the necessary filters
+func (di *DriverInterface) SetupFlowHandle() error {
+	h, err := di.GenerateDriverHandle()
+	if err != nil {
+		return err
+	}
+	dh, err := NewDriverHandle(h, FlowHandle)
+	if err != nil {
+		return err
+	}
+
+	// Prepare handle filters for each interface
+	err = dh.createDriverHandleFilters()
+	if err != nil {
+		return err
+	}
+
+	// Set handle filters
+	err = dh.setFilters()
+	if err != nil {
+		return err
+	}
+
+	di.driverFlowHandle = dh
+	return nil
+}
+
+// SetupStatsHandle generates a windows Driver Handle, and creates a DriverHandle struct
+func (di *DriverInterface) SetupStatsHandle() error {
+	h, err := di.GenerateDriverHandle()
+	if err != nil {
+		return err
+	}
+
+	dh, err := NewDriverHandle(h, StatsHandle)
+	if err != nil {
+		return err
+	}
+
+	di.driverStatsHandle = dh
+	return nil
+
+}
+
+// GenerateDriverHandle creates a new windows handle attached to the driver
+func (di *DriverInterface) GenerateDriverHandle() (windows.Handle, error) {
+	p, err := windows.UTF16PtrFromString(di.path)
+	if err != nil {
+		return windows.InvalidHandle, err
+	}
+	log.Debug("Creating Driver handle...")
+	h, err := windows.CreateFile(p,
+		windows.GENERIC_READ|windows.GENERIC_WRITE,
+		windows.FILE_SHARE_READ|windows.FILE_SHARE_WRITE,
+		nil,
+		windows.OPEN_EXISTING,
+		0,
+		windows.Handle(0))
+	if err != nil {
+		return windows.InvalidHandle, err
+	}
+	log.Info("Connected to driver and handle created")
```

Comment:
we should get rid of this line or change it to debug/add extra context since it gets called several times

Reply:
Going to remove the line, since if we ever run into issues the error handling should return the information we need anyway.